src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs

```diff
@@ -85,7 +85,30 @@ public Type TriggerValueType
 
             public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             {
-                EventGridEvent triggerValue = value as EventGridEvent;
+                // convert value to EventGridEvent, extract {data} as JObject
+                EventGridEvent triggerValue = null;
+                if (value is string)
+                {
+                    try
+                    {
+                        triggerValue = JsonConvert.DeserializeObject<EventGridEvent>((string)value);
```

Comment:
>(string)value [](start = 85, length = 13)

nit: this is doing a cast stwice (line 90 and 94). usually we have a stylecop warning that would catch this.  Try:
string strValue =  value as string; 
if (strValue != null) { }

Reply:
or better yet (now):
`if (value is string stringValue)`